libraries/aws_iam_group.rb

```diff
@@ -0,0 +1,54 @@
+require '_aws'
+
+class AwsIamGroup < Inspec.resource(1)
+  name 'aws_iam_group'
+  desc 'Verifies settings for AWS IAM Group'
+  example "
+    describe aws_iam_group('mygroup') do
+      it { should exist }
+    end
+  "
+
+  include AwsResourceMixin
+  attr_reader :group_name
+
+  def to_s
+    "IAM Group #{group_name}"
+  end
+
+  private
+
+  def validate_params(raw_params)
+    validated_params = check_resource_param_names(
+      raw_params: raw_params,
+      allowed_params: [:group_name],
+      allowed_scalar_name: :group_name,
+      allowed_scalar_type: String,
+    )
+
+    if validated_params.empty?
+      raise ArgumentError, 'You must provide a group_name to aws_iam_group.'
+    end
+
+    validated_params
+  end
+
+  def fetch_from_aws
+    backend = AwsIamGroup::BackendFactory.create
+
+    begin
+      @aws_group_struct = backend.get_group(group_name: group_name)[:group]
```

Comment:
Please don't store the response as an instance variable.  The issue here is that you're creating an "inner object" that holds some or all of the state.  `self` is already an object with properties and methods - when we learn state, we should set it on `self`, via instance variables.

This is harmless here, since we're only looking at `exists`.  But it's a pattern I keep seeing...